### PR TITLE
Catty-232: Added tests for VariablesContainer (part 3)

### PR DESCRIPTION
**Pull request by neel-makhecha**
Added tests for the following methods:
- getUserVariableNamed
- getUserListNamed
- removeUserVariableNamed
- removeUserListNamed

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catty/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [ ] Include the name of the Jira ticket in the PR’s title
- [ ] Verify that the Jira ticket is in the status *Ready for Development*
- [ ] Include a summary of the changes plus the relevant context
- [ ] Choose the proper base branch (*develop*)
- [ ] Confirm that the changes follow the project’s coding guidelines
- [ ] Verify that the changes generate no compiler or linter warnings
- [ ] Perform a self-review of the changes
- [ ] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [ ] Confirm that new and existing unit tests pass locally
- [ ] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [ ] Stick to the project’s git workflow (rebase and squash your commits)
- [ ] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/CattyTests/DataModel/VariablesContainerTest.swift [hunk 1]
```diff
@@ -365,4 +365,124 @@ final class VariablesContainerTest: XCTestCase {
...
+
+        let container = VariablesContainer()
+
+        let result = container.addObjectVariable(userVariable1, for: objectA)
+        XCTAssertTrue(result)
+
+        container.programVariableList.add(userVariable2 as Any)
+
+        XCTAssertFalse(container.removeUserVariableNamed(userVariable1?.name, for: objectB))
+        XCTAssertTrue(container.removeUserVariableNamed(userVariable1?.name, for: objectA))
```
Comment (m-herold) [inline src/CattyTests/DataModel/VariablesContainerTest.swift:449]: Could you please also verify that the number of variables in the container decreased by one and that the proper variable has been removed?

File: src/CattyTests/DataModel/VariablesContainerTest.swift [hunk 2]
```diff
@@ -365,4 +365,124 @@ final class VariablesContainerTest: XCTestCase {
...
+
+        let container = VariablesContainer()
+
+        let result = container.addObjectList(list1, for: objectA)
+        XCTAssertTrue(result)
+
+        container.programListOfLists.add(list2 as Any)
+
+        XCTAssertFalse(container.removeUserListNamed(list1?.name, for: objectB))
+        XCTAssertTrue(container.removeUserListNamed(list1?.name, for: objectA))
```
Comment (m-herold) [inline src/CattyTests/DataModel/VariablesContainerTest.swift:480]: Could you please also verify that the number of lists in the container decreased by one and that the proper list has been removed?

**Comment by neel-makhecha** [comment]
Hi Michael, made changes as suggested. Please review.

**Comment by m-herold** [comment]
Please rebase, resolve merge conflicts and squash your commits.

**Pull request merged**